### PR TITLE
[WIP] Declaring new functions

### PR DESCRIPTION
**Pull request by adweex**
## Description

Created a few new files and functions before i remove old ones. These are just placeholders. Commands are also moved to their own file for better readability

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Comment by Molicheu** [comment]
A lot of this looks like it was derived from this resource https://github.com/tclrd/tclrd_vehiclekeys, It pretty much does the same thing

**Comment by adweex** [comment]
> A lot of this looks like it was derived from this resource https://github.com/tclrd/tclrd_vehiclekeys, It pretty much does the same thing

Never heard of that resource. In the past 1.5 years I've been developing my own server and resources. During this time i used qb for half a year and qbx for a year, sadly i had lot of issues with vehiclekeys because not every client receives keys (out of 1/20 someone will not get any). I've seen this suggetstion:
https://github.com/Qbox-project/qbx_vehiclekeys/issues/4

So i contacted Manason that i would rewrite the vehiclekeys using statebags maybe later on with key item support and other extras. I would never ever attempt to steal someone elses code (specially that my quest system is not long ago got stolen by my "friend"). It might look the same, and might have the same functionality but the code is written by me 100%. I only used qbx_vehiclekeys as the base of the script. But i am open to suggestions.

**Comment by adweex** [comment]
Well that is not for me to decide. I'll wait then until there is some kind of resolution in that regard.

**Comment by Manason** [comment]
> Well that is not for me to decide. I'll wait then until there is some kind of resolution in that regard.

It kinda is your decision my man lol. You are the one taking ownership of vehicle keys. I don't think anyone else is going to make the decision. Do you want to continue down your current path or take code from the other resource?

**Comment by adweex** [comment]
> > Well that is not for me to decide. I'll wait then until there is some kind of resolution in that regard.
> 
> It kinda is your decision my man lol. You are the one taking ownership of vehicle keys. I don't think anyone else is going to make the decision. Do you want to continue down your current path or take code from the other resource?

I'd rather develop my own code than write a bridge for someone else's. I also wouldn't use that code if there's another way, mainly because I find it easier to maintain mine. I see QBX as a team effort, so I believe it's not solely my decision. I'm open to suggestions; for example you, Manason, have much more knowledge than I do. Making such decision feels like a significant responsibility. However, it's possible that TCLRD's code is better in some ways than I write. I've checked it, and it looks good, but I don't really use "OOP" in Lua (if that coding on the server is oop). It just feels weird to me (does it have any advantage at all?). Give me some insight, what would you do? (sorry i am not used to such conversations :D)

**Comment by Manason** [comment]
> > > Well that is not for me to decide. I'll wait then until there is some kind of resolution in that regard.
> > 
> > 
> > It kinda is your decision my man lol. You are the one taking ownership of vehicle keys. I don't think anyone else is going to make the decision. Do you want to continue down your current path or take code from the other resource?
> 
> I'd rather develop my own code than write a bridge for someone else's. I also wouldn't use that code if there's another way, mainly because I find it easier to maintain mine. I see QBX as a team effort, so I believe it's not solely my decision. I'm open to suggestions; for example you, Manason, have much more knowledge than I do. Making such decision feels like a significant responsibility. However, it's possible that TCLRD's code is better in some ways than I write. I've checked it, and it looks good, but I don't really use "OOP" in Lua (if that coding on the server is oop). It just feels weird to me (does it have any advantage at all?). Give me some insight, what would you do? (sorry i am not used to such conversations :D)

I think it's fine to stick with your current plan and improve qbx_vehiclekeys

File: server/main.lua [hunk 1]
```diff
@@ -89,70 +89,77 @@ function HasKeys(id, plate)
...
-            optional = true
-        },
-    },
-    restricted = false,
-}, function (source, args)
-    local src = source
-    TriggerClientEvent('qb-vehiclekeys:client:GiveKeys', src, args.id)
+--- ##########################
+--- ### SERVER-ONLY EVENTS ###
+--- ##########################
```
Comment (MafewTM) [inline server/main.lua:94]: these organizational comments arent really needed

**Comment by MafewTM** [comment]
Additionally if youre going to have file organization for the functions, you dont really need to name them clientFunctions or serverFunctions if they are in the client and server folders

**Comment by adweex** [comment]
> Additionally if youre going to have file organization for the functions, you dont really need to name them clientFunctions or serverFunctions if they are in the client and server folders

Manason said the same thing. Only named them clientFunctions for better readability and to make search easier. While i code many windows are open at the same time. If i have 3 functions.lua then it can confuse me. But I'll do the changes.

**Pull request merged**